### PR TITLE
Update repo to use dusk-plonk v0.2.0

### LINKED ISSUE #38: Port Poseidon to PLONK v0.2.0
@vlopes11 could take care of the encryption scheme part while @CPerezz takes care about the Merkle-Tree related stuff.

### PR DESCRIPTION
**Pull request by CPerezz**
Closes #38

### CODE REVIEW

**Comment by CPerezz** [comment]
We can bump up the version once #34 is closed by @vlopes11

File: src/tree/mod.rs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 use std::io;
 
-use dusk_bls12_381::Scalar;
+use dusk_plonk::bls12_381::Scalar;
```
Comment (vlopes11) [inline src/tree/mod.rs]: We renamed everywhere else to `BlsScalar` but not here. Why?

Comment (CPerezz) [reply src/tree/mod.rs]: Because I missed it hahah. Good catch!

**Review by vlopes11 — APPROVED** [review]
lgtm, in the future we may want to export `Scalar` from `dusk_plonk` as either `BlsScalar` or `JubJubScalar`

**Comment by CPerezz** [comment]
> lgtm, in the future we may want to export `Scalar` from `dusk_plonk` as either `BlsScalar` or `JubJubScalar`

It is already done like this in the prelude. But you need to explicitly mention it when you deal with the re-exported crate types. There's nothing we can do. 

And I don't think it makes sense to re-export both crates and sepparately, both `Scalar` structures with the type alias.

**Comment by CPerezz** [comment]
It seems that due to the LTO optimizations done, there's a regression on nigtly and the linker can't work correctly. Therefore, the code is not being compiled and exits with an error related to missing symbols.

See: https://github.com/rust-lang/wg-cargo-std-aware/issues/53

Since it's only affecting nightly, I'll merge and see what happens with the next nightly release.

**Pull request merged**